diff --git a/settings.c b/settings.c
index a00af07..7ce1c32 100644
--- a/settings.c
+++ b/settings.c
@@ -2236,6 +2236,7 @@ PROGMEM static const setting_detail_t setting_detail[] = {
      { Setting_SubroutineOptions, Group_General, "Subroutine options", NULL, Format_Bitfield, "Prescan for internal M98 subroutines", NULL, NULL, Setting_IsExtendedFn, set_suboptions, get_int, is_setting_available }
 };
 
+#ifndef NO_SETTINGS_DESCRIPTIONS
 PROGMEM static const setting_descr_t setting_descr[] = {
     { Setting_PulseMicroseconds, "Step pulse length in microseconds.\\n"
                                  "Minimum depends on the processor and is typically in the range of 1 - 2.5.\\n\\n"
@@ -2450,6 +2451,7 @@ PROGMEM static const setting_descr_t setting_descr[] = {
     { Setting_MotorFaultsInvert, "Invert motor fault inputs" }
 */
 };
+#endif
 
 static setting_details_t setting_details = {
     .is_core = true,
@@ -2457,8 +2459,13 @@ static setting_details_t setting_details = {
     .n_groups = sizeof(setting_group_detail) / sizeof(setting_group_detail_t),
     .settings = setting_detail,
     .n_settings = sizeof(setting_detail) / sizeof(setting_detail_t),
+#ifndef NO_SETTINGS_DESCRIPTIONS
     .descriptions = setting_descr,
     .n_descriptions = sizeof(setting_descr) / sizeof(setting_descr_t),
+#else
+    .descriptions = NULL,
+    .n_descriptions = 0,
+#endif
     .save = settings_write_global
 };
 
